lib/explorer/data_frame.ex

```diff
@@ -4048,6 +4048,19 @@ defmodule Explorer.DataFrame do
   When working with grouped dataframes, be aware that only groups from the first
   dataframe are kept in the resultant dataframe.
 
+  Note that for lazy frames it's not possible to check the number of rows, so
+  you need to pass the option `:check_n_rows` as `false`.
+
+  ## Options
+
+    * `:check_n_rows` - Controls if all dataframes must have the same number of rows.
```

Comment:
One option is to move the logic for checking rows inside the backend. So eager will check it and lazy won't. Here we simply document that lazy backends won't check the number of rows and it is up to you to guarantee they match if you don't want data loss. WDYT?